Lesson 2

bwengals opened the pull request.

bwengals committed:
add start of lesson 2

canyon289 committed:
Update lesson 2

canyon289 committed:
Update lesson 2

canyon289 committed:
More updates

canyon289 committed:
Expand lesson 2 Section 10 and 20

canyon289 committed:
Merge pull request #2 from bwengals/section_10_script

Section 10 script

canyon289 committed:
Add back bills origin notebook

bwengals committed:
add bit of code with least squares ests, a little at the end

bwengals committed:
add code to plot a basic GP fit

bwengals committed:
add periodic gp

canyon289 commented on lesson_2.ipynb:
```diff
@@ -0,0 +1,2129 @@
+{
```
<p>Great point at how connecting the dots is not a good approach. Also if our method was to connect the dots how would we estimate uncertainty?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='108'/>

canyon289 replied:
<pre class="ql-syntax" spellcheck="false">Line #1.    ### HIDE THIS CODE
</pre><p>Great idea</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='109'/>

canyon289 replied:
<p>Great idea plotting just a section of the data, focuses on just one part of the trend which is great and wont overwehlm people too much. You also picked a great part of the series to showcase these as the third point from the left is "out of band"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='113'/>

canyon289 commented:
I continue to be more excited every update I see

For my own sanity and convenience relinking our script here
https://github.com/IntuitiveBayes/gp_course_internal/blob/main/scripts/lesson_02.md

bwengals committed:
add most of lesson 30, a bit more to go

bwengals committed:
finish rough of section 30

bwengals committed:
save updates

canyon289 committed:
Add expanded Lesson 02

canyon289 committed:
Update Lesson 02

canyon289 committed:
More edits

canyon289 committed:
Update lesson 02 expanded

canyon289 committed:
Update Lesson02 expanded

bwengals committed:
Merge branch 'lesson02' of github.com:bwengals/etc into lesson02

bwengals committed:
fix lengthscale, fix matplotlib obj oriented, clean up mauna loa example more

bwengals commented:
@canyon289 Went ahead and pushed the latest updates to this branch.  Fixed the kernel equations, fixed the matplotib obj oriented stuff, and cleaned up mauna loa example some more.

canyon289 commented:
Nits on specific slides. Some of this is one me, im just listing it here
**Update**: @bwengals ignore this comment block as its all on Section 10 and 20, I realized this is likely because the hide code extension is not enabled on your computer. Please do read the next comment below for fixes on section 30

## Find a bigger mauna loa picture
This one looks fairly boring, would be nice to get a more tropical one

![image](https://user-images.githubusercontent.com/7213793/181272129-d46406f9-0c0b-49d3-aa18-b09357c56ea3.png)

## Fix image being cut off
![image](https://user-images.githubusercontent.com/7213793/181272535-c36f2b9f-ccb7-4e6a-b9f8-b438201d8444.png)

## Resize plot
![image](https://user-images.githubusercontent.com/7213793/181272606-3fc72ca0-323c-42ec-9d10-26d41bc16c5b.png)

## Fix PyMC Image
![image](https://user-images.githubusercontent.com/7213793/181272847-338a5b4f-bb49-4121-87d1-b34cef36eca8.png)

## Hide Plotting code 
![image](https://user-images.githubusercontent.com/7213793/181273078-66aa900a-2d18-49fb-90c1-d0144bbaf221.png)

## Hide Plotting code
![image](https://user-images.githubusercontent.com/7213793/181273249-7adc14ed-9f25-44ac-9068-a9648322cc94.png)

## Fix slides here
![image](https://user-images.githubusercontent.com/7213793/181273350-0e0c07e9-d978-4307-9c07-b29189a89a10.png)

## In section 10 and 20 fix up most slides
I'll stop commenting on each one

canyon289 commented:
Heres comments for Section 40. Please do fix these

## Bigger farmhouse picture
![image](https://user-images.githubusercontent.com/7213793/181274013-c3d5ec40-2c8d-4476-accb-e84e61767805.png)

## Fix image size to fit into 1920x1080
![image](https://user-images.githubusercontent.com/7213793/181274184-2bf962e6-7941-40fa-8ddb-49f4985e4d5e.png)

## Rehide the code blocks
Please ensure the hide code block extension is enabled on your computer. Thes images are sized correctly, but if the extension is not there itll shift the image. See the comparison below

![image](https://user-images.githubusercontent.com/7213793/181274257-338a2e01-3f2d-4d96-b52d-6866d83c27a0.png)
![image](https://user-images.githubusercontent.com/7213793/181274499-721016e7-50eb-41d6-9ef5-7db99397fbbe.png)

## Fix this slide
![image](https://user-images.githubusercontent.com/7213793/181274789-ea39f170-d725-45b1-a3e7-814ae387e0dc.png)

canyon289 commented:
In NB extensions please ensure you have
* Hide Input
* isort formatter
* Jupyterblack
* Split cells notebook

all enabled
![image](https://user-images.githubusercontent.com/7213793/181275318-4d4c3fcb-848b-4cb8-9dcf-24010e949980.png)

bwengals commented:
<p>good catch, lots of other plots missing these too </p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>Yup looks like it is, deleted this cell</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>thanks!</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>gotcha</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>fixed it up a bit</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>It was added back everywhere but here whoops.  I added a bit of text about the 2, too</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>I see what you're saying, but I think "short" and "long" is the typical nomenclature.  I think the intent of that is actually to conflate lengthscale with distance, like "short lengthscale" means "short distance effects".  There's nothing wrong with small, medium, large, but I'd resist standardizing one way or the other though since I think both are common "out there"</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>Yeah that's a cool example, I'll add that.  </p><p> </p><p>The lengthscale idea can be that general and more.  You "can" do things like a mahalanobis distance, where the lengthscales can have these sort of correlations between each other.  Never seen anyone do it.  </p><p> </p><p>I have seen where say x is very high dimensional, you can do a PCA like thing here too</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>added</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>good call, pretty much added this text</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>Looks good to me.  Added a bulled that we've defined kernel functions, what they mean, what they represent.</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals commented:
<p>For sure.  I'm not sure about text/narrative here, since we're not building this up.  How do these sorts of demos go in the intro course?</p>

---

 <a href='https://app.reviewnb.com/bwengals/etc/pull/1/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

bwengals committed:
a couple small style edits

bwengals committed:
another pass

canyon289 committed:
Add rise.css

canyon289 commented:
Just a heads up
1. I added rise.css so youll need to repull this
2. Some of the slides metadata seems to be missing now, such as slide indicators, and hiding code cells. Those will need to be readded for things to work correctly, this can be done at the end though once all the content is in

canyon289 committed:
Fix up lesson 2 slides

bwengals committed:
remove subslides

canyon289 committed:
Add section 50

canyon289 committed:
Update section 50

canyon289 committed:
Merge pull request #9 from bwengals/lesson2_section_50

Lesson2 section 50

bwengals commented:
I guess I looked at 50 but forgot to leave comments.  I think it looks great.  I think we should merge this branch and then go from there

canyon289 commented:
Awesome, sounds like this entire lesson is "performance ready" then. Lets focus now then on that aspect in the coming week!

canyon289 committed:
Minor updates for sizing

canyon289 committed:
Rerun notebook

canyon289 committed:
Update notebook

canyon289 committed:
Update lesson02

canyon289 committed:
Add back residuals

canyon289 committed:
Update lesson02

canyon289 committed:
Update Section 40

canyon289 committed:
More updates to section 50

canyon289 committed:
Update section 50

canyon289 committed:
Update lesson 02

canyon289 committed:
Update lesson 02

canyon289 committed:
Update lesson 02

bwengals commented:
should we merge this?  should rename the notebooks first though.  there are a couple others in here that are out of date

canyon289 commented:
Instead of merging, when were done we just copy  over the relevant files into the private repo and close this PR. Eventually were going to delete this whole repo, its just here temporarily so we can use reviewnb while in development

canyon289 commented:
To be clear Im happy to copy it over! Ill do it later today as well

canyon289 committed:
Add section recap

canyon289 committed:
Add lesson recap and pictures

canyon289 committed:
Add additional lesson 2 expanded changes

canyon289 committed:
Slight modifications to lesson 02

canyon289 committed:
Update Lesson 02

canyon289 committed:
Fix Gaussian process

canyon289 commented:
Superseded by https://github.com/IntuitiveBayes/GaussianProcessesAlpha
